### PR TITLE
Get credit score

### PR DESCRIPTION
**Pull request by SteveOye**

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -27,6 +28,7 @@ class _MyAppState extends State<MyApp> {
   bool isLoading = false;
   bool _response = false;
   // bool _statementResponse = false;
```
Comment (zabeehpericulum) [inline example/lib/main.dart]: comment

File: lib/flutter_periculum.dart [hunk 2]
```diff
@@ -83,46 +85,44 @@ class FlutterPericulum {
     return exisitingStatementResponse;
   }
 
-  // static Future<dynamic> statementTransaction(
-  //     {required String token, required String key}) async {
-  //   Map<String, dynamic> myResponse;
+  // static Future<TransactionStatementResponse> statementTransaction({
```
Comment (zabeehpericulum) [inline lib/flutter_periculum.dart]: remove comments please

File: lib/flutter_periculum.dart [hunk 3]
```diff
@@ -83,46 +85,44 @@ class FlutterPericulum {
...
+    required String token,
+    required String statementKey,
+  }) async {
+    String response = await _channel.invokeMethod('getExistingCreditScore', {
+      'token': token,
+      'statementKey': statementKey,
+    });
+
+    try {
+      List<ExisitingCreditResponse> responseList;
```
Comment (zabeehpericulum) [inline lib/flutter_periculum.dart]: `CreditScoreResponse`

File: lib/models/TransactionStatementResponse.dart [hunk 4]
```diff
@@ -0,0 +1,64 @@
+// To parse this JSON data, do
```
Comment (zabeehpericulum) [inline lib/models/TransactionStatementResponse.dart]: no comments please.

File: lib/models/TransactionStatementResponse.dart [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+import 'dart:convert';
+
+TransactionStatementResponse transactionStatementResponseFromJson(String str) =>
+    TransactionStatementResponse.fromJson(json.decode(str));
+
+String transactionStatementResponseToJson(TransactionStatementResponse data) =>
+    json.encode(data.toJson());
+
+class TransactionStatementResponse {
```
Comment (zabeehpericulum) [inline lib/models/TransactionStatementResponse.dart]: `TransactionResponse`

File: lib/models/TransactionStatementResponse.dart [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        data: List<Datum>.from(json["data"].map((x) => Datum.fromJson(x))),
+      );
+
+  Map<String, dynamic> toJson() => {
+        "status": status,
+        "data": List<dynamic>.from(data.map((x) => x.toJson())),
+      };
+}
+
+class Datum {
```
Comment (zabeehpericulum) [inline lib/models/TransactionStatementResponse.dart]: `Transaction`

File: lib/models/TransactionStatementResponse.dart [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    json.encode(data.toJson());
+
+class TransactionStatementResponse {
+  TransactionStatementResponse({
+    required this.status,
+    required this.data,
+  });
+
+  bool status;
+  List<Datum> data;
```
Comment (zabeehpericulum) [inline lib/models/TransactionStatementResponse.dart]: `List<Transaction> transactions;`

File: lib/flutter_periculum.dart [hunk 8]
```diff
@@ -83,46 +85,44 @@ class FlutterPericulum {
...
+
+  //   var result = json.decode(response) as List;
+  //   print(result[0]);
+  //   map = json.decode(result[0]);
+  //   TransactionStatementResponse transactionStatementResponse =
+  //       TransactionStatementResponse.fromJson(map);
+  //   return transactionStatementResponse;
   // }
 
+  static Future<List<ExisitingCreditResponse>> getExisitingCreditScore({
```
Comment (zabeehpericulum) [inline lib/flutter_periculum.dart]: `getExistingCreditScoresForStatementKey`

**Review by zabeehpericulum — CHANGES_REQUESTED** [review]

File: example/lib/main.dart [hunk 9]
```diff
@@ -21,12 +22,11 @@ class MyApp extends StatefulWidget {
 
 class _MyAppState extends State<MyApp> {
   String _mobileData = 'Unknown';
-  // String _platformVersion = 'Unknown';
   late ExisitingStatementResponse exisitingStatementResponse;
   late AffordabilityResponse affordabilityResponse;
   bool isLoading = false;
   bool _response = false;
-  // bool _statementResponse = false;
+  late List<ExisitingCreditScoreResponse> creditResponse;
```
Comment (zabeehpericulum) [inline example/lib/main.dart]: I thought we changed the name of this to CreditScoreResponse (The model)

**Review by zabeehpericulum — CHANGES_REQUESTED** [review]

File: lib/flutter_periculum.dart [hunk 10]
```diff
@@ -84,7 +84,7 @@ class FlutterPericulum {
     return exisitingStatementResponse;
   }
 
-  static Future<List<ExisitingCreditScoreResponse>> getExisitingCreditScore({
+  static Future<List<CreditScoreResponse>> getExisitingCreditScore({
```
Comment (SteveOye) [inline lib/flutter_periculum.dart:87]: It fixed here

File: lib/models/TransactionStatementResponse.dart [hunk 11]
```diff
@@ -0,0 +1,59 @@
+import 'dart:convert';
+
+TransactionStatementResponse transactionStatementResponseFromJson(String str) =>
+    TransactionStatementResponse.fromJson(json.decode(str));
+
+String transactionStatementResponseToJson(TransactionStatementResponse data) =>
+    json.encode(data.toJson());
+
+class TransactionStatementResponse {
```
Comment (zabeehpericulum) [inline lib/models/TransactionStatementResponse.dart]: StatementTransactionsResponse

**Review by zabeehpericulum — APPROVED** [review]

**Comment by zabeehpericulum** [comment]
Please rename Statement.dart to StatementResponse.dart.

**Pull request merged**